openfeature/client.go

```diff
@@ -41,6 +41,7 @@ type Client struct {
 	metadata          ClientMetadata
 	hooks             []Hook
 	evaluationContext EvaluationContext
+	name              string
```

Comment:
```suggestion
	domain              string
```

`name` has been renamed to `domain` in the most recent spec. I think there's a few places we need to update param names, etc, but if we're adding new fields, prefer `domain`.

Reply:
yes, thx i will rename that one. For the other ocurrences. I just saw that the old/wrong terminology is used in quite a lot of places. Wdyt about if I just open another PR that contains all the renaming?

Reply:
You can open a PR or even just an issue. :pray: